ap.md

```diff
@@ -0,0 +1,256 @@
+
+{% highlight javascript %}
+// These two dependencies remain the same
+var tessel = require('tessel');
+var http = require('http');
+
+// Require two other core Node.js modules
+var fs = require('fs');
+var url = require('url');
+
+var server = http.createServer(function (request, response) {
+  // Here is the biggest change
+  var urlParts = url.parse(request.url, true); // Break up the url into easier-to-use parts
+  var ledRegex = /leds/; // Create a regular expression to match requests to toggle LEDs
+
+  if (urlParts.pathname.match(ledRegex)) {
+    // If there is a request containing the string 'leds' call a function, toggleLED
+    toggleLED(urlParts.pathname, request, response);
+  } else {
+    // All other request will call a function, showIndex
+    showIndex(urlParts.pathname, request, response);
+  }
+});
+
+// Stays the same
+server.listen(8080);
+
+// Stays the same
+console.log('Server running at http://192.168.1.101:8080/');
+{% endhighlight %}
+
+Inside our `createServer` callback function, we've replaced the old plain text response with a basic router that catches requests to '/leds/:index' and defaults all other requests to a separate function. Let's add those functions below the `console.log`:
+
+{% highlight javascript %}
+// Respond to the request with our index.html page
+function showIndex (url, request, response) {
+  response.writeHead(200, {"Content-Type": "text/html"});
+  // Use fs to read in index.html
+  fs.readFile(__dirname + '/index.html', function (err, content) {
+    if (err) {
+      throw err;
+    }
+
+    // Serve the content of index.html read in by fs
+    response.end(content);
+  });
+}
+
+// Toggle the led specified in the url and respond with its state
+function toggleLED (url, request, response) {
+  var indexRegex = /(\d)$/; // Create a regular expression to find the number at the end of the url
+  var result = indexRegex.exec(url); // Capture the number, returns an array
+  var index = +result[1]; // Grab the captured result from the array and make sure it's a Number
+
+  var led = tessel.led[index]; // Use the index to refence the correct LED
+
+  // Toggle the state of the led and call the callback after that's done
+  led.toggle(function (err) {
+    if (err) {
+      // Log the error and send back a 500 (internal server error) response to the client
+      console.log(err);
+      response.writeHead(500, {"Content-Type": "application/json"});
+      response.end(JSON.stringify({error: err}));
+    } else {
+      // The led was successfully toggled, so respond with the state of the toggled led using led.isOn
+      response.writeHead(200, {"Content-Type": "application/json"});
+      response.end(JSON.stringify({on: led.isOn}));
+    }
+  });
+}
+{% endhighlight %}
+
+One last step is to create a `.tesselinclude` file in the project directory and add any files not required by `ap.js` that we want to deploy with it. For this project, that means adding one line: 
```

Comment:
Let's give a bit more explanation on why this is needed. Something like:

In order to make code pushing more efficient, Tessel only pushes the entry point file and its Node dependencies by default. Since `index.html` is not included in this default push, we'll need to explicitly require it with a `.tesselinclude` file.

Create a new file called `.tesselinclude` and copy and paste the following: